projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid.component.ts

```diff
@@ -2060,6 +2059,24 @@ export class IgxPivotGridComponent extends IgxGridBaseDirective implements OnIni
         return cols;
     }
 
+    private sortColumnFields(fields: Map<string, any>, expressions: any): Map<string,any> {
```

Comment:
Might be simpler to just sort the original data before the Map is generated, like we do for the rows in the pivot pipes. Something like:

```
protected autogenerateColumns() {
...
        const expressions = PivotSortUtil.generateDimensionSortingExpressions(this.columnDimensions);
        const sortedData = DataUtil.sort(cloneArray(data, true), expressions, this.sortStrategy, this);
```

And then just use the sorted data when generating the unique values. They should be generated in the correct order, without the need to sort them again later.

            fieldsMap = PivotUtil.getFieldsHierarchy(
                sortedData,
                this.columnDimensions,
                PivotDimensionType.Column,
                this.pivotKeys
            );

Or at least they will be consistent with how the rows are sorted.
In both case for dates there will be issues since they get sorted as strings and not as actual date object. But I don't think that can be resolved without introducing some custom displayValue function or custom compare function.